### ISSUE TITLE
[Doc]: the matplotlib notebook magic requires `notebook<7`

### ISSUE THREAD

**Issue by qci-amos** [matplotlib/matplotlib#27144]
### Documentation Link

https://matplotlib.org/stable/users/explain/figure/interactive.html#jupyter-notebooks-jupyterlab

### Problem

As discussed in this issue:
https://github.com/jupyter/notebook/issues/7115

it sounds like the wording in the documentation should be clarified because a user installing unpinned dependencies will run into a problem with old notebooks.

### Suggested improvement

Instead of:

> If you only need to use the classic notebook, you can use

it could say

> If you only need to use the classic notebook (i.e., `notebook<7`), you can use

(I'd also suggest a better error message somewhere)

**Comment by jklymak** [comment]
That seems reasonable, though I find that whole section somewhat convoluted.  Probably should be fixed in figure intro as well.  Any interest in making a PR?

**Comment by qci-amos** [comment]
I'm not qualified to make a PR! 😄 I'm just wearing a devops hat here.

**Comment by mattymo30** [comment]
I would like to take a stab at this, at the very least fixing the wording suggested above.

**Comment by qci-amos** [comment]
An important detail to us that we've noticed is that with `notebook==7` the fix:

```
%matplotlib widget
```
isn't sufficient for us. We're interested in more than just basic interactivity (e.g., panning, zooming) but also live updating, for example, the plot changing as data streams in. This is the closest issue to our problem that we've found: https://github.com/matplotlib/ipympl/issues/290


Probably this isn't relevant to the PR in progress; I'm just making a note of functionality that has changed between versions 6 and 7 of `notebook`.
